src/backend/access/external/hd_work_mgr.c

```diff
@@ -768,6 +769,12 @@ make_allocation_output_string(List *segment_fragments)
 	initStringInfo(&segwork);
 	appendStringInfoString(&segwork, SEGWORK_PREFIX);
 	
+	/* Add dfs_address from pg_filespace to the segment data. Fixes HAWQ-462 *//* dfs_address from pg_filespace entry */
+	char* dfs_address = NULL;
+	get_hdfs_location_from_filespace(&dfs_address);
```

Comment:
probably add check to see if it's secure or not. enable_secure_filesystem == true